compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -915,4 +1010,69 @@ private boolean isMemberAllowed(SymbolEnv env, BSymbol symbol) {
                 && env.enclInvokable.symbol.receiverSymbol.type.tsymbol == symbol.owner
                 || isMemberAllowed(env.enclEnv, symbol));
     }
+
+    /**
+     * Returns the eligibility to use 'stamp' inbuilt function against the respective expression.
+     *
+     * @param targetType target type that 'stamp' function is used
+     * @return eligibility to use 'stamp' function
+     */
+    private boolean isStampSupportedForTargetType(BType targetType) {
+        if (!types.isAnydata(targetType)) {
+            return false;
+        }
+
+        switch (targetType.tag) {
+            case TypeTags.ARRAY:
+                //Primitive type array does not support stamp because primitive arrays aren't using ref registry.
+                int arrayConstraintTypeTag = ((BArrayType) targetType).eType.tag;
+                return !(arrayConstraintTypeTag == TypeTags.INT || arrayConstraintTypeTag == TypeTags.BOOLEAN ||
+                        arrayConstraintTypeTag == TypeTags.FLOAT || arrayConstraintTypeTag == TypeTags.BYTE ||
+                        arrayConstraintTypeTag == TypeTags.STRING || arrayConstraintTypeTag == TypeTags.DECIMAL);
+            case TypeTags.UNION:
+            case TypeTags.INT:
+            case TypeTags.BOOLEAN:
+            case TypeTags.STRING:
+            case TypeTags.FLOAT:
+            case TypeTags.DECIMAL:
+            case TypeTags.BYTE:
+            case TypeTags.TABLE:
+                return false;
+            default:
+                return true;
+        }
+    }
+
+    /**
+     * Returns the eligibility whether stamp can be on the given value type.
+     *
+     * @param sourceType source type used for the stamp operation
+     * @return eligibility to use as the target type for 'stamp' function
+     */
+    private boolean isStampSupportedForSourceType(BType sourceType) {
+
```

Comment:
Shall we remove the extra line added after the function signature?

Reply:
Fixed.